qtranslate_core.php

```diff
@@ -165,11 +165,12 @@ function qtranxf_detect_language( &$url_info ) {
     // parse language and front info from HTTP_REFERER
     if ( isset( $_SERVER['HTTP_REFERER'] ) && $parse_referrer ) {
         $http_referer             = $_SERVER['HTTP_REFERER'];
+        $http_request_uri         = $_SERVER['REQUEST_URI'];
         $url_info['http_referer'] = $http_referer;
 
         // if needed, detect front- vs back-end
         $parse_referrer_language = true;
-        if ( strpos( $http_referer, '/wp-admin' ) !== false ) {
+        if ( strpos( $http_request_uri, '/wp-admin' ) !== false || strpos( $http_referer, '/wp-admin' ) !== false ) {
```

Comment:
That part of the code is dedicated to parsing HTTP_REFERER ($parse_referrer = true). Not sure we should look at the request URI here.

Reply:
@mweimerskirch this check should probably not be in the if-statement regarding HTTP_REFERER. How can we reproduce the problem you encountered?

Reply:
My bad, I missed a few things... I'll check this again better.